server/midddleware/ddtrace.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package middleware
+
+import (
+	"context"
+	"strings"
+
+	middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/tigrisdata/tigris/server/metrics"
+	"github.com/tigrisdata/tigris/server/request"
+	ulog "github.com/tigrisdata/tigris/util/log"
+	"google.golang.org/grpc"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func getMethodName(fullMethod string) string {
+	return strings.Split(fullMethod, "/")[2]
```

Comment:
This might be prone to formatting changes and likely more Tigris than tracing specific at the moment. Should we handle this a level above in "Tigris context"?

Reply:
I guess the whole file is very tigris specific. Based on the name I expected datadog-specific tracing here. Should we just rename the file to tracing.go?

Reply:
I refactored this so it uses the same things as the counters, so all the tagging logic is in one place now.